### DISCUSSION TITLE
Desktop uploader?

### DISCUSSION THREAD

**Discussion by Stooovie** [immich-app/immich#9966] [General]
It would be great to have an uploader on desktop. I use iCloud Photos (55k photos) as my main photo library and right now, I have to use an iPad as an uploader device as the iPhone gets bogged down for minutes when used for uploading. Even in the latest M4 iPad, it takes Immich almost four minutes to evaluate what to sync. Inherent iOS/iPadOS limitations make this a bit unwieldy and unreliable. The initial upload took almost a month, and not due to slow internet. 

Dedicated desktop uploader could help. Ideas?

**Comment by bo0tzz** [comment]
(partial) duplicate of #5049.

For a large upload from a PC, you can use the CLI https://immich.app/docs/features/command-line-interface

**Reply by Stooovie** [reply]
Does it work with iCloud Photos Library that is in cloud by default?
